File: specs/020_test_epi_rct_ketamine_meth/scripts/00_check_env.R
# 00_check_env.R -- Verify R toolchain and package availability
# Task 20: Zed R/Python toolchain verification
# Run: Rscript scripts/00_check_env.R

cat("==== R Environment Check ====\n")
cat("R version:        ", R.version$version.string, "\n")
cat("Platform:         ", R.version$platform, "\n")
cat("Running under:    ", R.version$os, "\n\n")

cat("==== Library Paths ====\n")
for (p in .libPaths()) cat("  ", p, "\n")
cat("\n")

cat("==== Package Availability ====\n")
pkgs <- c(
  "tidyverse", "dplyr", "readr", "ggplot2",
  "survival", "gtsummary", "broom",
  "mice", "knitr", "rmarkdown",
  "languageserver", "styler", "lintr"
)

results <- data.frame(package = pkgs, installed = NA, loads = NA,
                      stringsAsFactors = FALSE)
for (i in seq_along(pkgs)) {
  pkg <- pkgs[i]
  results$installed[i] <- requireNamespace(pkg, quietly = TRUE)
  if (results$installed[i]) {
    ok <- tryCatch({
      suppressPackageStartupMessages(library(pkg, character.only = TRUE))
      TRUE
    }, error = function(e) FALSE)
    results$loads[i] <- ok
  } else {
    results$loads[i] <- FALSE
  }
  status <- if (results$loads[i]) "OK" else "MISSING"
  cat(sprintf("  %-20s %s\n", pkg, status))
}

cat("\n==== Summary ====\n")
cat("Available:", sum(results$loads), "/", nrow(results), "\n")
cat("Missing:  ", paste(results$package[!results$loads], collapse = ", "), "\n")

File: specs/archive/020_test_epi_rct_ketamine_meth/scripts/00_check_env.py
"""00_check_env.py -- Verify Python toolchain and module availability.

Task 20: Zed R/Python toolchain verification
Run: python3 scripts/00_check_env.py
"""

from __future__ import annotations

import importlib
import sys

print("==== Python Environment Check ====")
print(f"Python version:  {sys.version}")
print(f"Executable:      {sys.executable}")
print(f"Platform:        {sys.platform}")
print()

print("==== Module Availability ====")
modules = [
    "numpy",
    "pandas",
    "scipy",
    "matplotlib",
    "seaborn",
    "statsmodels",
    "sklearn",
    "pyarrow",
]

results: dict[str, bool] = {}
for mod in modules:
    try:
        importlib.import_module(mod)
        results[mod] = True
        print(f"  {mod:<15} OK")
    except ImportError:
        results[mod] = False
        print(f"  {mod:<15} MISSING")

print()
print("==== Summary ====")
present = sum(results.values())
total = len(results)
print(f"Available: {present} / {total}")
missing = [m for m, ok in results.items() if not ok]
if missing:
    print(f"Missing:   {', '.join(missing)}")
